torch/_higher_order_ops/__init__.py

```diff
@@ -19,4 +20,6 @@
     "flex_attention_backward",
     "hints_wrapper",
     "PrimHOPBase",
+    "foreach_map",
```

Comment:
nit - should we keep this private until it's more fully featured ?

Reply:
I kind of feel like it's ready tbh, it can handle arbitrary funcs today (just inductor may/may not fuse)

Reply:
torch._higher_order_ops is private, I wouldn't worry about it